CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs

```diff
@@ -0,0 +1,378 @@
+            item.StudentGroupId = request.StudentGroupId ?? item.StudentGroupId;
+            item.ThemeId = request.ThemeId ?? item.ThemeId;
+            item.EventStart = request.EventStart.HasValue ? new DateTime(item.EventStart.Year, item.EventStart.Month, item.EventStart.Day,
+                request.EventStart.Value.Hour, request.EventStart.Value.Minute, request.EventStart.Value.Second) : item.EventStart;
+            item.EventFinish = request.EventEnd.HasValue ? new DateTime(item.EventFinish.Year, item.EventFinish.Month, item.EventFinish.Day,
+                request.EventEnd.Value.Hour, request.EventEnd.Value.Minute, request.EventEnd.Value.Second) : item.EventFinish;
+
+            return item;
+        }
+
+        private async Task<string> ValidateCreateScheduleRequestAsync(CreateScheduleDto request)
+        {
+            if (request == null)
+            {
+                return "Request must not be null";
+            }
+
+            string error = null;
+
+            if (request.Pattern.Interval <= 0)
+            {
+                error = "Interval value out of range";
+            }
+
+            if (!await _unitOfWork.StudentGroupRepository.IsEntityExistAsync(request.Context.GroupID))
+            {
+                error = "Group does not exist";
+            }
+
+            if (request.Context.MentorID.HasValue && !await _unitOfWork.MentorRepository.IsEntityExistAsync(request.Context.MentorID.Value))
+            {
+                error = "Mentor does not exist";
+            }
+
+            if (request.Context.ThemeID.HasValue && !await _unitOfWork.ThemeRepository.IsEntityExistAsync(request.Context.ThemeID.Value))
+            {
+                error = "Theme does not exist";
+            }
+
+            switch (request.Pattern.Type)
+            {
+                case PatternType.Daily:
+                    break;
+                case PatternType.Weekly:
+                    if (request.Pattern.DaysOfWeek == null || request.Pattern.DaysOfWeek.Count == 0)
+                    {
+                        error = "Target days not provided";
+                    }
+                    break;
+                case PatternType.AbsoluteMonthly:
+                    if (request.Pattern.Dates == null || request.Pattern.Dates.Count == 0)
+                    {
+                        error = "Target dates not provided";
+                    }
+                    break;
+                case PatternType.RelativeMonthly:
+                    if ((request.Pattern.DaysOfWeek == null || request.Pattern.DaysOfWeek.Count == 0) 
+                        || (request.Pattern.Index != null ? request.Pattern.Index <= 0 : false))
+                    {
+                        error = "Target days not provided";
+                    }
+                    break;
+                default:
+                    error = "Pattern type not supported";
+                    break;
+            }
+
+            return error;
+        }
```

Comment:
these checks will override previous messages.
as example if all of them are valid "error" message will contain only the last one message.

or return in each "if" "case"
or if you want to do all checks and accumulate all messages than use StringBuilder.

Reply:
accumulating errors is a good idea